### PR TITLE
Zero inputs Concat fix and Scatter Update ops fix

### PR DESCRIPTION
**Pull request by jane-intel**
### Details:
 - *Enabled data2vec_audio model*

### Tickets:
 - *95831*

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp [hunk 1]
```diff
@@ -718,6 +719,30 @@ pass::EliminateEltwise::EliminateEltwise() {
     this->register_matcher(m, callback);
 }
 
+pass::EliminateScatterUpdate::EliminateScatterUpdate() {
+    MATCHER_SCOPE(EliminateScatterUpdate);
+    auto scatter_pattern =
+        pattern::wrap_type<opset8::ScatterUpdate, opset8::ScatterNDUpdate, opset8::ScatterElementsUpdate>();
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp:725]: nitpick: I would recommend to use the latest opset (opset10)

File: src/common/transformations/src/transformations/common_optimizations/remove_concat_zero_dim_input.cpp [hunk 2]
```diff
@@ -45,7 +45,19 @@ ov::pass::RemoveConcatZeroDimInput::RemoveConcatZeroDimInput() {
 
         bool inputs_removed = concat->get_input_size() > concat_inputs.size();
         if (inputs_removed) {
-            concat->set_arguments(concat_inputs);
+            if (concat_inputs.empty()) {
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/remove_concat_zero_dim_input.cpp:48]: but the problem with empty input tensors to concat op still exists, right?
I mean the situation when not all tensors to concat are empty but some of them and they are cut. In some cases these empty tensors affect concat output shape. 
do we plan to fix this?

Comment (jane-intel) [reply src/common/transformations/src/transformations/common_optimizations/remove_concat_zero_dim_input.cpp:48]: I don't plan to fix this in current release as we know there could be drawbacks. As @slyalin proposed -- let's fix this model in current release and plan axis fix for future releases

Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/remove_concat_zero_dim_input.cpp:48]: ok, could you create a jira ticket?

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/remove_concat_zero_dim_input.cpp (hunk 2 above)
Comment (jane-intel) [reply src/common/transformations/src/transformations/common_optimizations/remove_concat_zero_dim_input.cpp:48]: Done

**Pull request merged**